node/src/mpc_client.rs

```
@@ -299,14 +299,18 @@ impl MpcClient {
         mpc_client: Arc<MpcClient>,
     ) -> anyhow::Result<()> {
         let mut tasks = AutoAbortTaskCollection::new();
-        loop {
-            let channel = channel_receiver.recv().await.unwrap();
+        while let Some(channel) = channel_receiver.recv().await {
             let mpc_clone = mpc_client.clone();
             tasks.spawn_checked(
                 &format!("passive task {:?}", channel.task_id()),
                 async move { mpc_clone.process_channel_task(channel).await },
             );
         }
+
+        tracing::info!(
+            "Network task channel receiver is closed. Exiting monitor_passive_channels_inner."
+        );
+        Ok(())
```

shouldn't we return an error here?

---

Good point. 

I don't have a strong opinion about the return type. If we also return an Error here, we might as well return `anyhow::Error` instead of a result.

Most important is that we don't have these unsafe unwraps.